LanternExtractor/EQ/Wld/Fragments/ObjectInstance.cs

```diff
@@ -55,8 +56,16 @@ class ObjectInstance : WldFragment
             // In main zone, it points to a 0x16 fragment
             // In objects.wld, it is 0
             int unknown2 = Reader.ReadInt32();
+            var x = Reader.ReadSingle();
+            var y = Reader.ReadSingle();
+            var z = Reader.ReadSingle();
 
-            Position = new vec3(Reader.ReadSingle(), Reader.ReadSingle(), Reader.ReadSingle());
+            // Sometimes we are getting the lowest value of signed int 16. Reset to 0 otherwise this z will be way below the world. Round for floating point errors.
+            if (Math.Round(z) <= short.MinValue)
```

Comment:
This change shouldn't be in this fragment. My guess is that they use a "dropping" solution where they place a bunch of trees and dropped them onto the terrain below. Some of them missed and end up at the bottom of the world. So the issue here is that these trees will now be positioned at z = 0. What I'd recommend is filtering them out in the export code. I do it in the importer in a rather lazy way. `if(z < 30000) continue;`

Can definitely handle it more elegantly.

Reply:
Good call here nobody else should care about dropped trees here, moved this to MeshObjWriter